src/_common/fireside/post/post-model.ts

```diff
@@ -321,6 +322,21 @@ export class FiresidePost extends Model implements ContentContainerModel {
 		return this.$_save(`/web/communities/manage/reject/${c.id}`, 'post');
 	}
 
+	$moveChannel(community: Community, channel: CommunityChannel) {
+		const c = this.getTaggedCommunity(community);
+		if (!c) {
+			throw new Error('Cannot move a post to a channel in a community it is not tagged in');
+		}
+
+		if (community.id !== channel.community_id) {
```

Comment:
This should not be possible, right? And we are catching this in backend anyway iirc.

Reply:
Correct, it shouldn't but since this is a function on the post model it by itself can be misused, so I put this guard.